src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -432,6 +432,25 @@ private void initializedOrRunning2ContributionTimeout(Task task) {
         }
     }
 
+    private void running2AllWorkersFailed(Task task) {
+        boolean isRunningTask = task.getCurrentStatus().equals(RUNNING);
+        boolean haveAllWorkerFailed = replicatesService.getReplicates(task.getChainTaskId())
+                .stream()
+                .filter(replicate -> ReplicateStatus.isFailure(replicate.getCurrentStatus()))
+                .count() == workerService.getAliveWorkers().size();
```

Comment:
1. IMO check is not accurate, because we can have:
worker1replicate *_failed  and not alive anymore
worker2replicate *_failed and alive
worker3replicate *_failed and alive
failed == 3 vs getAliveWorkers == 2

Maybe something like:
for each getAliveWorkers { get task.getReplicateOfWorker; if replicate.isFailure failed ++ }
if failed == aliveWorkers, return Status.ALL_FAILED
which checks all alive workers did participate and failed

2. In addition, it might be interested to return Status.ALL_FAILED_ON_SAME_ERROR and Status.ALL_FAILED_FOR_VARIOUS_ERRORS.
A  Status.ALL_FAILED_ON_SAME_ERROR (if multiple workers) might express the issue isn't caused by workers themselves, but rather by App, Requester..

Reply:
1. What about lost workers who contributed ? I think we should first check if at least one worker is in `COMPUTED`, `CONTRIBUTING` or `CONTRIBUTED` status, if it is the case even if the worker is lost we should wait.
2. I'm not sure we need this, it might be redundant since we already have the information under Replicates.

Reply:
1. New version has the following logic: a task is transitioned to `RUNNING_FAILED` if:
  1.1. All alive workers have tried to run the task;
  1.2 All workers that have tried to run the task have failed, whether they are alive or lost;
  1.3 No replicate has reached the `COMPUTED` status.

2. I agree with @zguesmi: this piece of information can be retrieved otherwise.